src/sort/bigrat.rs

```diff
@@ -48,23 +48,36 @@ impl Sort for BigRatSort {
         add_primitives!(eg, "to-f64" = |a: Q| -> f64 { a.to_f64().unwrap() });
 
         add_primitives!(eg, "pow" = |a: Q, b: Q| -> Option<Q> {
-            if a.is_zero() {
-                if b.is_positive() {
+            if !b.is_integer() {
+                // fractional powers are forbidden.
+                // reject this even for the zero case
+                None
+            } else if a.is_zero() {
+                // remove zero from the field of rationals
+                // so that multiplicative inverse is always safe
+                if b.is_zero() {
+                    // 0^0 = 1 by common convention
+                    Some(Q::one())
+                } else if b.is_positive() {
+                    // 0^n = 0 where (n > 0)
                     Some(Q::zero())
                 } else {
-                    None
-                }
-            } else if b.is_zero() {
-                Some(Q::one())
-            } else if let Some(b) = b.to_i64() {
-                if let Ok(b) = usize::try_from(b) {
-                    num::traits::checked_pow(a, b)
-                } else {
-                    // TODO handle negative powers
+                    // 0^n => (1/0)^(abs n) where (n < 0)
```

Comment:
Is this comment still relevent?

Reply:
Which part? The change removes the `TODO` but I add in a comment for which mathematical case the branch is handling